build-and-test-with-gradle/action.yaml

```
@@ -0,0 +1,113 @@
+name: Build and Test
+description: Builds and tests code using gradle
+
+inputs:
+  checkout-ref:
+    description: Ref to checkout from Git
+    default: ${{ github.event.pull_request.head.sha }}
+    required: false
+
+  github-token:
+    description: Token used for accessing GitHub.com
+    required: true
+  sonar-token:
+    description: "Token used for pushing test results to SonarCloud.io"
+    required: true
+  sonar-project-base-dir:
+    description: Use this property when you need the analysis to take place in a directory other than the one from which it was launched.
+    default: ./
+    required: false
+  sonar-organization:
+    description: The key of the organization to which the project belongs to
+    default: neohelden
+    required: false
+  sonar-projectKey:
+    description: The project's unique key. Allowed characters are letters, numbers, -, _, . and :, with at least one non-digit.
+    default: ${{ github.event.repository.name }}
+    required: false
+  sonar-branch-name:
+    description: Branch name to by analyzed. Must be empty for PR builds.
+    default: ${{ github.event.ref }}
+    required: false
+
+  default-branch:
+    description: Default development branch of this repository
+    default: ${{ github.event.repository.default_branch }}
+    required: false
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Branch name
+      id: extract_branch
+      run: |
+        BRANCH_NAME=${{ inputs.sonar-branch-name }}
+
+        if [ -n "$BRANCH_NAME" ]
+        then
+          echo "##[set-output name=branch;]-Dsonar.branch.name=$(echo ${BRANCH_NAME#refs/heads/})"
+        else
+          echo "##[set-output name=branch;]"
+        fi
+      shell: bash
+
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0 ## fix shallow clone detected by SonarCloud.io
+        ref: ${{ inputs.checkout-ref }}
+
+    - name: Set up JDK 11
+      uses: actions/setup-java@v3
+      with:
+        java-version: "11"
+        distribution: "adopt"
```

Lets change this to 17 :) 
```suggestion
    - name: Set up JDK 17
      uses: actions/setup-java@v3
      with:
        java-version: "17"
        distribution: "adopt"
```